src/net.rs

```diff
@@ -177,6 +180,57 @@ pub trait HostResponse {
     fn finish(&mut self) -> Result<(), Error>;
 }
 
+/// Represents a physical port that can be used to interact with client devices.
+///
+/// This for example is used for a PA RoT to send messages to a AC RoT.
+///
+/// This trait provides a generic mechanism for sending and receiving requests
+/// from devices.
+pub trait DevicePort<'a> {
```

Comment:
What does this lifetime indicate? I suspect you don't *actually* want it.

Reply:
It's required for borrowing the buffer.

Reply:
Why can't you accept `&[u8]` rather than giving it a fixed lifetime?

Reply:
Rust complains that the lifetimes don't match then for the implementation of `InMemDevice`

This is similar to the existing `request()` function: https://github.com/lowRISC/manticore/blob/master/src/net.rs#L285

Reply:
This is only necessary if you're going to take ownership of the buffer and save it, which doesn't seem to make a lot of sense for what is morally a blocking `write()` syscall? I would have expected the in-memory version to perform a copy (which it will have to do anyway if I find that this needs to take `Write` instead of a buffer.)